lib/OptimizationOptimisers/src/OptimizationOptimisers.jl

```diff
@@ -95,77 +95,73 @@ function SciMLBase.__solve(cache::OptimizationBase.OptimizationCache{
     gevals = 0
     t0 = time()
     breakall = false
-    begin
-        for epoch in 1:epochs
-            if breakall
-                break
+    progress_id = uuid4()
+    for epoch in 1:epochs, d in data
+        if cache.f.fg !== nothing && dataiterate
+            x = cache.f.fg(G, θ, d)
+            iterations += 1
+            fevals += 1
+            gevals += 1
+        elseif dataiterate
+            cache.f.grad(G, θ, d)
+            x = cache.f(θ, d)
+            iterations += 1
+            fevals += 2
+            gevals += 1
+        elseif cache.f.fg !== nothing
+            x = cache.f.fg(G, θ)
+            iterations += 1
+            fevals += 1
+            gevals += 1
+        else
+            cache.f.grad(G, θ)
+            x = cache.f(θ)
+            iterations += 1
+            fevals += 2
+            gevals += 1
+        end
+        opt_state = OptimizationBase.OptimizationState(
+            iter = iterations,
+            u = θ,
+            p = d,
+            objective = x[1],
+            grad = G,
+            original = state)
+        breakall = cache.callback(opt_state, x...)
+        if !(breakall isa Bool)
+            error("The callback should return a boolean `halt` for whether to stop the optimization process. Please see the `solve` documentation for information.")
+        elseif breakall
+            break
+        end
+        cache.progress &&
+            @info ProgressLogging.Progress(progress_id, iterations / maxiters;
+                name = "loss: $(round(first(first(x)); digits=3))")
```

Comment:
I don't think `@info` is needed here, as you can use a log level that does not print, but it's picked up by [`ProgressLogging`](https://julialogging.github.io/ProgressLogging.jl/stable/#ProgressLogging.progress-Tuple{Any}) integrations, such as the VS Code native progress bar. I think that `LogLevel(-1)` could be enough.